File: PFMPManager.Api/Models/Message.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PFMPManager.Api.Models
{
    // Represents a message sent in a PFMP chat
    [Table("Message")]
    public class Message
    {
        [Key]
        [Column("Id_Message")]
        public int Id_Message { get; set; }

        [Column("Id_PFMP")]
        public int Id_PFMP { get; set; }
   
        [Column("Id_Utilisateur")]
        public int Id_Utilisateur { get; set; }

        [Column("RoleExpediteur")]
        [MaxLength(50)]
        public string RoleExpediteur { get; set; } = string.Empty;

        [Column("Contenu")]
        [MaxLength(250)]
        public string Contenu { get; set; } = string.Empty;


        [Column("DateEnvoi")]
        public DateTime? DateEnvoi { get; set; }


    }
}

File: appli_pfmp/lib/model/message.dart
class MessagePfmp {
  final int idUtilisateur;
  final int idPfmp;
  final String roleExpediteur;
  final String contenu;
  final String? dateEnvoi;
  final int idMessage;

  const MessagePfmp({
    required this.idUtilisateur,
    required this.idPfmp,
    required this.roleExpediteur,
    required this.contenu,
    required this.dateEnvoi,
    required this.idMessage,
  });

  factory MessagePfmp.fromJson(Map<String, dynamic> json) => MessagePfmp(
    idUtilisateur: json['idUtilisateur'] ?? json['id_Utilisateur'] ?? 0,
    idPfmp: json['idPfmp'] ?? json['id_PFMP'] ?? 0,
    roleExpediteur: json['roleExpediteur'] ?? '',
    contenu: json['contenu'] ?? '',
    dateEnvoi: json['dateEnvoi'],
    idMessage: json['idMessage'] ?? json['id_Message'] ?? 0,
  );
}
